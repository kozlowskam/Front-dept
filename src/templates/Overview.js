import React from 'react';
import Intro from '../components/Intro';
import Card from '../components/Card';
import CardSmall from '../components/CardSmall';
import Quote from '../components/Quote';
import Brands from '../components/Brands';
import Form from '../components/Form';

export class Overwiev extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      activeItem: {},
      cases: [],
      clients: {}
    }
    this.overwievDiv = React.createRef();
  }

  componentDidMount() {

    if (this.props.item) {
      this.setState({
        activeItem: this.props.item,
        cases: this.props.item.content.cases,
        clients: this.props.item.content.clients
      });
    }
  }

  groupArr(arr, n) {
    let group = [];

    for (let i = 0, j = 0; i < arr.length; i++) {
      if (i >= n && i % n === 0)
        j++;
      group[j] = group[j] || [];
      group[j].push(arr[i])
    }

    return group;
  }

  getRandomIndex(max) {
    return Math.floor(Math.random() * Math.floor(max));
  }


  render() {
    const { activeItem, cases, clients } = this.state
    let casesBefore;
    let casesAfter;
    let randomCase;

    if (cases && cases.length > 0) {

      let withImg = cases.filter((item) => (
        item.image
      ))

      randomCase = withImg[this.getRandomIndex(withImg.length - 1)]

      withImg.map((item) => item.type = "img")
      let groupImg = this.groupArr(withImg, 5)

      let withoutImg = cases.filter((item) => (
        !item.image
      ))

      let groupnoImg = this.groupArr(withoutImg, 2)

      let newArr = groupImg.map((group, i) => {
        if (groupnoImg[i]) {
          let merged = [{
            type: "noimg",
            cases: groupnoImg[i]
          }]
          merged.type = "noimg"
          return group.concat(merged)
        } else {
          return group
        }
      })

      let casesArr = [].concat.apply([], newArr);

      casesBefore = (casesArr && casesArr.length > 0) ? (
        casesArr.slice(0, 10).map((item, i) => {
          switch (item.type) {
            case 'img':
              return <Card item={item} />
            case 'noimg':
              return <CardSmall item={item} />
            default:
              return null;
          }
        })
      ) : null

      casesAfter = (
        casesArr.slice(10, casesArr.length).map((item, i) => {
          switch (item.type) {
            case 'img':
              return <Card item={item} key={i} />
            case 'noimg':
              return <CardSmall item={item} key={i} />
            default:
              return null;
          }
        })
      )
    };

    const header = randomCase ? (
      <Intro item={randomCase} />
    ) : null

    const quote = activeItem.quote ? (
      <Quote item={activeItem.quote} />
    ) : null

    const clientsWrap = (clients && clients.brands && clients.brands.length > 0) ? (
      <Brands clients={clients} />
    ) : null

    return (
      <div className="b-overview">
        {header}
        <div className="b-overview__wrapper">
          {casesBefore}
          {quote}
          {casesAfter}
        </div>
        {clientsWrap}
        <Form />
      </div>
    )
  }
}

export default Overwiev;