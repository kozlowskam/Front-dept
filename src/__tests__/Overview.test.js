import React from "react";
import TestRender from "react-test-renderer";
import Overview from "../templates/Overview";


describe("Overview", () => {
  test("it maches", () => {
    const component = TestRender.create(<Overview />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});