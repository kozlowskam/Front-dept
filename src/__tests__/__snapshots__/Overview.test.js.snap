// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview it maches 1`] = `
<div
  className="b-overview"
>
  <div
    className="b-overview__wrapper"
  />
  <div
    className="c-form"
  >
    <div
      className="c-form__title"
    >
      <span>
        QUESTION? 
      </span>
      <span>
        WE ARE HERE TO HELP!
      </span>
    </div>
    <div
      className="c-form__wrapper"
    >
      <form
        className="c-form__form"
      >
        <label
          className="c-form__label"
        >
          <span>
            NAME
          </span>
          <input
            className="c-form__input"
            type="text"
          />
        </label>
        <label
          className="c-form__label"
        >
          <span>
            EMAIL
          </span>
          <input
            className="c-form__input"
            onChange={[Function]}
            pattern="^[a-zA-Z0-9.!#$%&'*+/=?^_\`{|}~-]+@[a-zA-Z0-9-]+(?:\\\\.[a-zA-Z0-9-]+)*$"
            type="email"
          />
          <span
            className="c-form__required"
          >
            this field is required
          </span>
        </label>
        <label
          className="c-form__label"
        >
          <span>
            MESSAGE
          </span>
          <input
            className="c-form__input"
            type="text"
          />
        </label>
      </form>
      <button
        className="c-form__button is-disabled"
        onClick={[Function]}
      >
        SEND
      </button>
    </div>
  </div>
</div>
`;
